docs/sources/reference/components/loki/loki.process.md

```diff
@@ -1848,6 +1850,82 @@ stage.timestamp {
 }
 ```
 
+### `stage.truncate`
+
+The `stage.truncate` inner block configures a processing stage that truncates log entries, labels, extracted map fields, or structured metadata that exceed configured limits.
+
+The `stage.truncate` block doesn't support any arguments and is only configured via a number of nested inner `rule` blocks.
+
+The following block is supported inside the definition of `stage.truncate`:
+
+| Block              | Description                   | Required |
+| ------------------ | ----------------------------- | -------- |
+| [`rule`][rule]     | Defines a truncation rule.    | yes      |
+
+[rule]: #rule
+
+#### `rule`
+
+Defines a truncation rule that will apply to the log line, labels, structured_metadata, or extracted map.
+
+The following arguments are supported:
+
+| Name          | Type     | Description                                                  | Default  | Required |
+|---------------|----------|--------------------------------------------------------------|----------|----------|
+| `limit`       | `string` | Maximum length before truncating.                            | `""`     | yes      |
+| `source`      | `string` | Source of the data to truncate. If empty, will truncate all. | `""`     | no       |
+| `source_type` | `string` | Source location of the data to truncate.                     | `"line"` | no       |
+| `suffix`      | `string` | Suffix to append to truncated values.                        | ``       | no       |
+
+The `limit` attribute should be expressed in logical units, for example `"1KiB"`.
+The stage checks the length of the log line, label values, or structured metadata values against the configured limit and truncates if it exceeds the limit.
```

Comment:
len(s) will return the size in bytes, not the number of characters... and some characters will take multiple bytes. So this is accurate, but I thought maybe we mention somewhat that this is length of encoded bytes we are talking about here.